src/mlpack/methods/bayesian_ridge/bayesian_ridge.cpp

```diff
@@ -0,0 +1,276 @@
+/**
+ * @file bayesian_ridge.cpp
+ * @author Clement Mercier 
+ *
+ * Implementation of Bayesian Ridge regression.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "bayesian_ridge.hpp"
+#include <mlpack/core/util/log.hpp>
+#include <mlpack/core/util/timers.hpp>
+
+using namespace mlpack;
+using namespace mlpack::regression;
+
+
+BayesianRidge::BayesianRidge(const bool centerData,
+                             const bool scaleData) :
+  centerData(centerData),
+  scaleData(scaleData)
+{/* Nothing to do */}
+
+double BayesianRidge::Train(const arma::mat& data,
+                            const arma::rowvec& responses)
+{
+  Timer::Start("bayesian_ridge_regression");
+
+  arma::mat phi;
+  arma::rowvec t;
+  arma::colvec eigval;
+  arma::mat eigvec;
+  arma::colvec eigvali;
+
+  // Preprocess the data. Center and scale.
+  CenterScaleData(data,
+                  responses,
+                  centerData,
+                  scaleData,
+                  phi,
+                  t,
+                  dataOffset,
+                  dataScale,
+                  responsesOffset);
+
+  // Compute this quantities once and for all.
+  const arma::colvec vecphitT = phi * t.t();
+  const arma::mat phiphiT =  phi * phi.t();
+
+  arma::eig_sym(eigval, eigvec, phiphiT);
+
+  // Detect singular matrix.
+  if (eigval[0] < 1e-8)
+  {
+    Log::Warn << "Singular matrix. Two lines or more are colinear."
+              <<  std::endl;
+    return -1;
+  }
+
+  unsigned short p = data.n_rows, n = data.n_cols;
+  // Initialize the hyperparameters and
+  // begin with an infinitely broad prior.
+  alpha = 1e-6;
+  beta =  1 / (var(t) * 0.1);
+
+  double tol = 1e-3;
```

Comment:
What do you think, should we make `tol, nIterMax` configurable by the user (add them to the constructor) and set some reasonable default values?

Reply:
I never changed them but of course we can. So, I will make them accessible for the user.